src/components/EmailNotifications/VerifyEmailAddress.js

```diff
@@ -216,6 +195,13 @@ const VerifyEmailAddress = React.memo(function VerifyEmailAddress({
 })
 
 function LegalTermsAndPolicy({ termsAccepted, setTermsAccepted, theme }) {
+  const handleOnChange = useCallback(
```

Comment:
I think the last comment might've been misunderstood.

What do you think changing this component's API such that an `onChange()` is passed into the component via its props? That way the responsibility of handling the state change is placed in the parent, and this component is completely agnostic to what happens when the checkbox's state is changed.